sql/postgresql/pggrammar.g4

```diff
@@ -0,0 +1,4294 @@
+	| alterstatsstmt
+	| altertsconfigurationstmt
+	| altertsdictionarystmt
+	| alterusermappingstmt
+	| analyzestmt
+	| callstmt
+	| checkpointstmt
+	| closeportalstmt
+	| clusterstmt
+	| commentstmt
+	| constraintssetstmt
+	| copystmt
+	| createamstmt
+	| createasstmt
+	| createassertionstmt
+	| createcaststmt
+	| createconversionstmt
+	| createdomainstmt
+	| createextensionstmt
+	| createfdwstmt
+	| createforeignserverstmt
+	| createforeigntablestmt
+	| createfunctionstmt
+	| creategroupstmt
+	| creatematviewstmt
+	| createopclassstmt
+	| createopfamilystmt
+	| createpublicationstmt
+	| alteropfamilystmt
+	| createpolicystmt
+	| createplangstmt
+	| createschemastmt
+	| createseqstmt
+	| createstmt
+	| createsubscriptionstmt
+	| createstatsstmt
+	| createtablespacestmt
+	| createtransformstmt
+	| createtrigstmt
+	| createeventtrigstmt
+	| createrolestmt
+	| createuserstmt
+	| createusermappingstmt
+	| createdbstmt
+	| deallocatestmt
+	| declarecursorstmt
+	| definestmt
+	| deletestmt
+	| discardstmt
+	| dostmt
+	| dropcaststmt
+	| dropopclassstmt
+	| dropopfamilystmt
+	| dropownedstmt
+	| dropstmt
+	| dropsubscriptionstmt
+	| droptablespacestmt
+	| droptransformstmt
+	| droprolestmt
+	| dropusermappingstmt
+	| dropdbstmt
+	| executestmt
+	| explainstmt
+	| fetchstmt
+	| grantstmt
+	| grantrolestmt
+	| importforeignschemastmt
+	| indexstmt
+	| insertstmt
+	| listenstmt
+	| refreshmatviewstmt
+	| loadstmt
+	| lockstmt
+	| notifystmt
+	| preparestmt
+	| reassignownedstmt
+	| reindexstmt
+	| removeaggrstmt
+	| removefuncstmt
+	| removeoperstmt
+	| renamestmt
+	| revokestmt
+	| revokerolestmt
+	| rulestmt
+	| seclabelstmt
+	| selectstmt
+	| transactionstmt
+	| truncatestmt
+	| unlistenstmt
+	| updatestmt
+	| vacuumstmt
+	| variableresetstmt
+	| variablesetstmt
+	| variableshowstmt
+	| viewstmt
+	| plsqlconsolecommand
+	//| pl_block
+	|
+	;
+
+plsqlconsolecommand:MetaCommand EndMetaCommand?
+;
+
+callstmt: CALL func_application;
+
+createrolestmt: CREATE ROLE roleid opt_with optrolelist;
+
+opt_with: WITH
+        //| WITH_LA
+        |
+        ;
+
+optrolelist: optrolelist createoptroleelem
+           |
+           ;
+
+alteroptrolelist: alteroptrolelist alteroptroleelem
+                |
+                ;
+
+alteroptroleelem: PASSWORD sconst
+                | PASSWORD NULL_P
+                | ENCRYPTED PASSWORD sconst
+                | UNENCRYPTED PASSWORD sconst
+                | INHERIT
+                | CONNECTION LIMIT signediconst
+                | VALID UNTIL sconst
+                | USER role_list
+                | identifier
+;
+createoptroleelem: alteroptroleelem
+                 | SYSID iconst
+                 | ADMIN role_list
+                 | ROLE role_list
+                 | IN_P ROLE role_list
+                 | IN_P GROUP_P role_list
+;
+createuserstmt: CREATE USER roleid opt_with optrolelist;
+
+alterrolestmt: ALTER ROLE rolespec opt_with alteroptrolelist
+             | ALTER USER rolespec opt_with alteroptrolelist
```

Comment:
It's better to merge common subparts for performance and clarity.

```suggestion
alterrolestmt
    : ALTER ( ROLE rolespec opt_with alteroptrolelist
            | USER rolespec opt_with alteroptrolelist )
    ;
```

Reply:
For Phase 1 I'm trying to keep code as close to original grammar as possible, it makes it easy to keep track of original grammar
I think at 1st it should be proven that grammar  works correctly and only after that - optimization / etc